jerry-core/ecma/operations/ecma-objects-general.c

```diff
@@ -658,77 +656,50 @@ ecma_op_general_object_define_own_property (ecma_object_t *obj_p, /**< the objec
   ecma_property_types_t current_type = ECMA_PROPERTY_GET_TYPE (current_p);
   const bool is_current_data_descriptor = (current_type == ECMA_PROPERTY_TYPE_NAMEDDATA);
   const bool is_current_accessor_descriptor = (current_type == ECMA_PROPERTY_TYPE_NAMEDACCESSOR);
+  const bool is_current_configurable = ecma_is_property_configurable (current_p);
 
   JERRY_ASSERT (is_current_data_descriptor || is_current_accessor_descriptor);
 
-  bool is_every_field_in_desc_also_occurs_in_current_desc_with_same_value = true;
-  if (property_desc_p->is_value_defined)
+  if (!is_current_configurable)
   {
-    if (!is_current_data_descriptor
-        || !ecma_op_same_value (property_desc_p->value,
-                                ecma_get_named_data_property_value (current_p)))
+    if (property_desc_p->is_configurable
```

Comment:
Moving step 7 here (btw, comment not transfered) introduces an incompatible change. Let's assume the following situation (written in pseudo-code for brewity):

!current.configurable && prop.configurable
&& current.writable == prop.writable
&& current.value == prop.value
&& current.getter == prop.getter
&& current.setter == prop.setter

The original version will return with true, while this PR rejects it.